Source/WebCore/xml/XPathUtil.cpp

```diff
@@ -73,5 +74,10 @@ bool isValidContextNode(Node& node)
     return false;
 }
 
+bool isXMLSpace(UChar character)
+{
+    return character <= ' ' && (character == ' ' || character == '\t' || character == '\r' || character == '\n');
```

Comment:
This is equivalent to `return character == ' ' || character == '\t' || character == '\r' || character == '\n'`.

Reply:
This is a frequent performance optimization we have in our whitespace functions though. I'd keep as-is

Reply:
Surprised if the compiler doesn't optimize it away. Have we ever confirmed that it changes generated code?

Reply:
Just verified out of curiosity and llvm does generate the same code with -O2 (tried both arm64 and x86_64):
```
isXMLSpace1(char):                       // @isXMLSpace1(char)
        mov     x8, #9728                       // =0x2600
        and     w9, w0, #0xff
        movk    x8, #1, lsl #32
        cmp     w9, #33
        cset    w9, lo
        lsr     x8, x8, x0
        and     w0, w9, w8
        ret
isXMLSpace2(char):                       // @isXMLSpace2(char)
        mov     x8, #9728                       // =0x2600
        and     w9, w0, #0xff
        movk    x8, #1, lsl #32
        cmp     w9, #33
        cset    w9, lo
        lsr     x8, x8, x0
        and     w0, w9, w8
        ret
```

@Ahmad-S792 Let's simplify the code then.

Reply:
Will do in next push.

Reply:
Done.